packages/typespec-ts/src/transform/transformApiVersionInfo.ts

```diff
@@ -52,16 +48,20 @@ export function transformApiVersionInfo(
   };
 }
 
-function getOperationApiVersion(
+export function getOperationApiVersion(
   client: SdkClient,
-  program: Program,
   dpgContext: SdkContext
 ): ApiVersionInfo | undefined {
   const apiVersionTypes = new Set<string>();
   const locations = new Set<ApiVersionPosition>();
   const clientOperations = listOperationsInOperationGroup(dpgContext, client);
+  let hasApiVersionInClient = true;
+  let hasApiVersionInOperation = true;
   for (const clientOp of clientOperations) {
-    const route = ignoreDiagnostics(getHttpOperation(program, clientOp));
+    hasApiVersionInOperation = false;
```

Comment:
can we remove this and set hasApiVersionInOperation = false in line 59? Will it be the same or do you need this here?

Reply:
I think we will need to refresh it before we handle each operation.